xarray/tests/test_dataset.py

```diff
@@ -5849,6 +5849,18 @@ def test_binary_op_join_setting(self) -> None:
             actual = ds1 + ds2
             assert_equal(actual, expected)
 
+    def test_binary_ops_keep_attrs(self) -> None:
```

Comment:
I'd probably use `parametrize` for this: that way, we can differentiate between a failing `keep_attrs=True` and `keep_attrs=False`.

Something like:
```suggestion
@pytest.mark.parametrize(
    ["keep_attrs", "expected"],
    (
        (False, {}),
        (True, {"foo": "value1", "bar": "value3"}),
        ("drop_conflicts", {"bar": "value3"}),
    ),
    ids=lambda params: params[0],
)
def test_binary_ops_keep_attrs(self, keep_attrs, expected) -> None:
```
of course, the `attrs` of the `Dataset` objects can't be equal now, and the checks would also have to be adapted